northd/northd.c

```diff
@@ -2826,6 +2829,20 @@ join_logical_ports(const struct sbrec_port_binding_table *sbrec_pb_table,
                         arp_proxy, op->nbsp->name);
                 }
             }
+
+            /* Only used for the router type LSP whose peer is l3dgw_port */
+            if (smap_get(&op->nbsp->options, "enable_router_port_acl")) {
+                if (op->peer && is_l3dgw_port(op->peer)) {
+                    op->enable_router_port_acl = smap_get_bool(&op->nbsp->options,
+                                              "enable_router_port_acl", false);
+                } else {
+                    static struct vlog_rate_limit rl =
+                        VLOG_RATE_LIMIT_INIT(5, 1);
+                    VLOG_WARN_RL(&rl, "enable_router_port_acl option is not "
```

Comment:
How important is this warning ?

In this code we are calling smap_get twice.  This could be expensive on a large scale deployment
and we can avoid it.  I think we can ignore this option if its not a distributed gateway port.

How about just something like ?

if (op->peer && is_l3dgw_port(op->peer)) {
                  op->enable_router_port_acl = smap_get_bool(&op->nbsp->options,
                                                                                                    "enable_router_port_acl", false);
}